src/infra/core/src/ingest/fetch_service.rs

```diff
@@ -523,9 +523,13 @@ impl FetchService {
 
         let has_more = new_has_more_data_path.exists();
 
-        if target_path.metadata().int_err()?.len() == 0
-            && prev_source_state == source_state.as_ref()
-        {
+        if source_state.is_some() && source_state.as_ref() == prev_source_state {
```

Comment:
Minor: we can remove `source_state.is_some()` (the condition branch), since `as_ref()` already contains this branch

Reply:
The idea here is: we consider the source "up-to-date" if its state did not advance (e.g. ETH block number stayed the same).

We can have a situation of "uncacheable" source that doesn't have a state - in this case empty output data should be considered an actual valid output - `empty source  != up-to-date`.

So `is_some()` is important to ensure that `source_state` and `prev_source_state` are not `None`.